generators/server/templates/src/main/java/package/config/MicroserviceSecurityConfiguration.java.ejs

```diff
@@ -207,51 +207,13 @@ import org.zalando.problem.spring.web.advice.security.SecurityProblemSupport;
 @EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true)
 public class SecurityConfiguration extends ResourceServerConfigurerAdapter {
```

Comment:
This file must be excluded for the generation in file.js for OAuth2 gateway
=> the gateway is not a "Resource Server" and this class can conflict with the OAuth2SsoConfiguration that host the real spring security configuration for OAuht2 Gateway

Note : MicroserviceSecurityConfiguration.java.ejs is used by uaa authenticationType in both gateway and microservice ... (despite the name of the template)

For me, the condition in file.js for MicroserviceSecurityConfiguration.java.ejs should be changed from : 
` condition: generator => !(generator.applicationType !== 'microservice' && !(generator.applicationType === 'gateway' && (generator.authenticationType === 'uaa' || generator.authenticationType === 'oauth2')))`
to 
` condition: generator => generator.applicationType === 'microservice' || generator.authenticationType === 'uaa'`